From 2eacc7978613aab7dab63768097d2a6335221ec0 Mon Sep 17 00:00:00 2001
From: Daniel Ammann <daniel.ammann@bytesatwork.ch>
Date: Tue, 28 Apr 2020 14:41:06 +0200
Subject: [PATCH] ARM: dts: stm32: bytedevkit: Add tweaks introduced by st

This fixes ethernet for byteDEVKIT V1.1 and ethernet rx for V1.0. tx is not
working on V1.0 for now.
---
 arch/arm/boot/dts/stm32mp157c-bytedevkit-v1-0.dts | 8 +++++++-
 arch/arm/boot/dts/stm32mp157c-bytedevkit-v1-1.dts | 8 +++++++-
 2 files changed, 14 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/stm32mp157c-bytedevkit-v1-0.dts b/arch/arm/boot/dts/stm32mp157c-bytedevkit-v1-0.dts
index c7fcb4be0ab2..55eb60f303cc 100644
--- a/arch/arm/boot/dts/stm32mp157c-bytedevkit-v1-0.dts
+++ b/arch/arm/boot/dts/stm32mp157c-bytedevkit-v1-0.dts
@@ -38,7 +38,6 @@
 				 <STM32_PINMUX('C', 2, AF11)>, /* ETH_RGMII_TXD2 */
 				 <STM32_PINMUX('E', 2, AF11)>, /* ETH_RGMII_TXD3 */
 				 <STM32_PINMUX('G', 11, AF11)>, /* ETH_RGMII_TX_CTL */
-				 <STM32_PINMUX('A', 2, AF11)>, /* ETH_MDIO */
 				 <STM32_PINMUX('C', 1, AF11)>; /* ETH_MDC */
 			bias-disable;
 			drive-push-pull;
@@ -46,6 +45,13 @@
 		};
 
 		pins2 {
+			pinmux = <STM32_PINMUX('A', 2, AF11)>; /* ETH_MDIO */
+			bias-disable;
+			drive-push-pull;
+			slew-rate = <0>;
+		};
+
+		pins3 {
 			pinmux = <STM32_PINMUX('C', 4, AF11)>, /* ETH_RGMII_RXD0 */
 				 <STM32_PINMUX('C', 5, AF11)>, /* ETH_RGMII_RXD1 */
 				 <STM32_PINMUX('H', 6, AF11)>, /* ETH_RGMII_RXD2 */
diff --git a/arch/arm/boot/dts/stm32mp157c-bytedevkit-v1-1.dts b/arch/arm/boot/dts/stm32mp157c-bytedevkit-v1-1.dts
index ecc3e48ed37f..886ecd01dfc0 100644
--- a/arch/arm/boot/dts/stm32mp157c-bytedevkit-v1-1.dts
+++ b/arch/arm/boot/dts/stm32mp157c-bytedevkit-v1-1.dts
@@ -38,7 +38,6 @@
 				 <STM32_PINMUX('C', 2, AF11)>, /* ETH_RGMII_TXD2 */
 				 <STM32_PINMUX('E', 2, AF11)>, /* ETH_RGMII_TXD3 */
 				 <STM32_PINMUX('G', 11, AF11)>, /* ETH_RGMII_TX_CTL */
-				 <STM32_PINMUX('A', 2, AF11)>, /* ETH_MDIO */
 				 <STM32_PINMUX('C', 1, AF11)>; /* ETH_MDC */
 			bias-disable;
 			drive-push-pull;
@@ -46,6 +45,13 @@
 		};
 
 		pins2 {
+			pinmux = <STM32_PINMUX('A', 2, AF11)>; /* ETH_MDIO */
+			bias-disable;
+			drive-push-pull;
+			slew-rate = <0>;
+		};
+
+		pins3 {
 			pinmux = <STM32_PINMUX('C', 4, AF11)>, /* ETH_RGMII_RXD0 */
 				 <STM32_PINMUX('C', 5, AF11)>, /* ETH_RGMII_RXD1 */
 				 <STM32_PINMUX('H', 6, AF11)>, /* ETH_RGMII_RXD2 */
-- 
2.20.1

